idioms/coercion-arguments.md

```diff
@@ -1,18 +1,18 @@
-# Using &str instead of &String
+# Using coercion for arguments
 
 ## Description
 
-It is usually recommended to prefer a `&str` type over a `&String` type for arguments.  In most cases `&str` has more flexibility.  A `&String` will coerce to a `&str` type when required -- however, the opposite is not true.  
+Usings a target of a deref coericion can increase the flexibility of your code when you are deciding which argument type to use for a function argument.  For example; using `&str` instead of a `&String`, or `&[T]` in preference of `&Vec<T>`, or even `&T` as opposed to a `&Box<T>`.  Not only that, you can avoid layers of indirection for those instances where the owned type already provides a layer of indirection, as can be illustrated in each of the previous examples.  For instance, a `String` has a layer of indirection, so a `&String` will have two layers of indrection.  We can avoid this by using `&str` instead, and letting `&String` coerce to a `&str` whenever the function is invoked.  And finally, using deref targets for a function can increase the number of viable inputs for your function which may increase its utility.
```

Comment:
I think the first and last sentences of this paragraph are saying the same thing? Feels like this needs a refactor -- I'd put the "more flexibility" and "fewer layers of indirection" points in separate paragraphs, at least.